Source/WebCore/Modules/mediastream/PeerConnectionBackend.cpp

```diff
@@ -707,6 +764,41 @@ void PeerConnectionBackend::deref() const
     m_peerConnection->deref();
 }
 
+String PeerConnectionBackend::generateJSONLogEvent(uintptr_t identifier, LogEvent&& logEvent, bool isForGatherLogs)
+{
+    ASCIILiteral type;
+    String event;
+    WTF::switchOn(WTFMove(logEvent), [&](MessageLogEvent&& logEvent) {
+        type = "event"_s;
+        StringBuilder builder;
+        builder.append("{\"message\":\""_s, logEvent.message, "\",\"payload\":\""_s);
+        if (logEvent.payload)
+            builder.append(WTF::base64EncodeToString(*logEvent.payload));
+        builder.append("\"}"_s);
+        event = builder.toString();
+    }, [&](StatsLogEvent&& logEvent) {
+        type = "stats"_s;
+        event = WTFMove(logEvent);
+    });
+
+    if (isForGatherLogs) {
+        UNUSED_PARAM(identifier);
+        UNUSED_VARIABLE(type);
+        return event;
+    }
+
+    auto timestamp = WTF::WallTime::now().secondsSinceEpoch().microseconds();
+    return makeString("{\"peer-connection\":\""_s, hex(identifier), "\",\"timestamp\":"_s, timestamp, ",\"type\":\""_s, type, "\",\"event\":"_s, event, '}');
+}
+
+void PeerConnectionBackend::emitJSONLogEvent(String&& event)
+{
+    if (!m_isJSONLogStreamingEnabled)
+        return;
+
+    m_logFile->println(WTFMove(event));
```

Comment:
This is WPE/GTK specific, maybe we should conditionally compile.
For macOS/iOS, we would probably prefer to call the callback given in `CommandLineAPIHost::gatherRTCLogs`.